references/common/capability.go

```diff
@@ -109,81 +109,97 @@ func InstallCapability(client client.Client, mapper discoverymapper.DiscoveryMap
 	}
 	tp.Source = &types.Source{RepoName: centerName}
 	defDir, _ := system.GetCapabilityDir()
+	fileContent, err := ioutil.ReadFile(filepath.Clean(filepath.Join(repoDir, tp.Name+".yaml")))
+	if err != nil {
+		return err
+	}
 	switch tp.Type {
 	case types.TypeComponentDefinition:
-		var cd v1beta1.ComponentDefinition
-		workloadData, err := ioutil.ReadFile(filepath.Clean(filepath.Join(repoDir, tp.Name+".yaml")))
+		err = InstallComponentDefinition(client, fileContent, ioStreams, tp)
 		if err != nil {
 			return err
 		}
-		if err = yaml.Unmarshal(workloadData, &cd); err != nil {
-			return err
-		}
-		cd.Namespace = types.DefaultKubeVelaNS
-		ioStreams.Info("Installing component capability " + cd.Name)
-		if tp.Install != nil {
-			tp.Source.ChartName = tp.Install.Helm.Name
-			if err = helm.InstallHelmChart(ioStreams, tp.Install.Helm); err != nil {
-				return err
-			}
-			err = addSourceIntoExtension(cd.Spec.Extension, tp.Source)
-			if err != nil {
-				return err
-			}
-		}
-		if cd.Spec.Workload.Type == "" {
-			tp.CrdInfo = &types.CRDInfo{
-				APIVersion: cd.Spec.Workload.Definition.APIVersion,
-				Kind:       cd.Spec.Workload.Definition.Kind,
-			}
-		}
-		if err = client.Create(context.Background(), &cd); err != nil && !apierrors.IsAlreadyExists(err) {
-			return err
-		}
 	case types.TypeTrait:
-		var td v1beta1.TraitDefinition
-		traitdata, err := ioutil.ReadFile(filepath.Clean(filepath.Join(repoDir, tp.Name+".yaml")))
+		err = InstallTraitDefinition(client, mapper, fileContent, ioStreams, tp)
 		if err != nil {
 			return err
 		}
-		if err = yaml.Unmarshal(traitdata, &td); err != nil {
-			return err
-		}
-		td.Namespace = types.DefaultKubeVelaNS
-		ioStreams.Info("Installing trait capability " + td.Name)
-		if tp.Install != nil {
-			tp.Source.ChartName = tp.Install.Helm.Name
-			if err = helm.InstallHelmChart(ioStreams, tp.Install.Helm); err != nil {
-				return err
-			}
-			err = addSourceIntoExtension(td.Spec.Extension, tp.Source)
-			if err != nil {
-				return err
-			}
-		}
-		if err = HackForStandardTrait(tp, client); err != nil {
+	case types.TypeScope:
+		// TODO(wonderflow): support install scope here
+	case types.TypeWorkload:
+		return fmt.Errorf("unsupported capability type %v", types.TypeWorkload)
+	}
+
+	success := plugins.SinkTemp2Local([]types.Capability{tp}, defDir)
+	if success == 1 {
+		ioStreams.Infof("Successfully installed capability %s from %s\n", capabilityName, centerName)
+	}
+	return nil
+}
+
+// InstallComponentDefinition will add a component into K8s cluster and install it's controller
+func InstallComponentDefinition(client client.Client, workloadData []byte, ioStreams cmdutil.IOStreams, tp types.Capability) error {
```

Comment:
```suggestion
func InstallComponentDefinition(client client.Client, workloadData []byte, ioStreams cmdutil.IOStreams, tp *types.Capability) error {
```
There seems to be a problem with function encapsulation, here the function parameters should use pointer `*types.Capability`

Reply:
that's my negligenc. fixed